sdlib/d/gc/arena.d

```diff
@@ -148,6 +148,55 @@ public:
 		return null;
 	}
 
+	bool resizeLarge(shared(ExtentMap)* emap, ref PageDescriptor pd,
+	                 size_t size) shared {
+		assert(pd.extent !is null, "Extent is null!");
+		assert(pd.extent.arenaIndex == index, "Invalid arena index!");
+		assert(isLargeSize(size));
+
+		if (pd.isSlab()) {
+			return false;
+		}
+
+		auto computedPageCount = alignUp(size, PageSize) / PageSize;
+		uint newPages = computedPageCount & uint.max;
+
+		assert(newPages == computedPageCount, "Unexpected page count!");
+
+		auto e = pd.extent;
+		auto oldSize = e.size;
+		auto oldPages = e.size / PageSize;
+
+		if (oldPages == newPages) {
+			return true;
+		}
+
+		auto pagesDelta = resizePages(e, newPages);
+
+		// Tried to resize, but failed:
+		if (pagesDelta == 0) {
+			return false;
+		}
+
+		auto sizeDelta = pagesDelta * PageSize;
+		auto newSize = oldSize + sizeDelta;
+		auto newE = e.at(e.address, newSize, e.hpd);
+
+		if (unlikely(!emap.remap(newE))) {
```

Comment:
Why remap the whole extent? In one case you only got to extend it (so this does a ton of needless work) and in the other you don't need to map anything.